crates/sui-config/src/genesis.rs

```diff
@@ -950,23 +950,28 @@ fn process_package(
         ctx.digest(),
         protocol_config,
     );
-    let package_id = ObjectID::from(*modules[0].self_id().address());
     let mut gas_status = SuiGasStatus::new_unmetered();
-    adapter::verify_and_link(
-        vm,
-        &temporary_store,
-        &modules,
-        package_id,
-        gas_status.create_move_gas_status(),
+    let module_bytes = modules
+        .into_iter()
+        .map(|m| {
+            let mut buf = vec![];
+            m.serialize(&mut buf).unwrap();
+            buf
+        })
+        .collect();
+    let pt = {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        builder.publish(module_bytes);
```

Comment:
@tzakian @amnn this will affect #8825. Maybe we should just have not generate upgrade caps in the case where you don't generate a package ID? 
We could rename `ExecutionMode::generate_package_id` to something like. `predefined_packages`?

Reply:
Yeah, I think that's fine -- we don't want upgrade caps for the frameworks.  I'm kind of fine with keeping the name as-is (it makes sense to me that if we are not generating package IDs, then we don't want to generate the UpgradeCap for that ID).

Reply:
I'll land the UpgradeCap stuff today, so it can be folded into this PR